proxy/pkg/zdmproxy/coalescer.go

```diff
@@ -137,7 +141,12 @@ func (recv *writeCoalescer) RunWriteQueueLoop() {
 					}
 
 					log.Tracef("[%v] Writing %v on %v", recv.logPrefix, f.Header, connectionAddr)
-					err := writeRawFrame(tempBuffer, connectionAddr, recv.shutdownContext, f)
+					var newf = f
+					if recv.frameProcessor != nil {
+						newf = f.Clone()
+						recv.frameProcessor.AssignUniqueId(newf)
+					}
+					err := writeRawFrame(tempBuffer, connectionAddr, recv.shutdownContext, newf)
```

Comment:
I've tried that initially but the tests fail if I don't clone the frame, I haven't determined exactly why yet, but I imagine it's because there are references being held somewhere else.

Reply:
I think it would be wise to understand where those failures are coming from, this `clone` call might just be hiding the issue.

Reply:
Actually, it might be a good idea to try and create a function that clones only the header part of the frame but doesn't clone the body and the frame processor should call that function as well as here.

Reply:
@joao-r-reis Yeah, I've been eager to investigate this deeper but I wanted to get the low-hanging fruits out of the way first.
I think I got to the bottom of this now. The problem is that the same `*frame.RawFrame` reference is passed to both origin and target cluster connectors, so with *no cloning* when the frameprocessor changes the frame's stream id, it changes it in both places.
I tried cloning just the Header struct as you suggested, but when we do `frame.Header = clonedHeader`, it will still change the reference in both places (because the parent is the same).
In the end I had to create a new parent `frame.RawFrame` struct and pass it the original Body with the cloned Header.